### ISSUE TITLE
Extract communicator logic from the web interface

### ISSUE THREAD

**Issue by dasanra** [safe-global/safe-pm#104]
# Part 1: Define the problem

## What problem are you trying to solve?
Currently is not possible to reuse the Safe Apps communicator logic in any other interface. Current app frame is tightly coupled to the web interface and would prevent flexibility at the time of migrating or reusing Safe Apps implementation in another frontend.

## What is your hypothesis?
If the logic is extracted to a separated library other projects building in top of the Safe could reuse this to integrate with Apps. Circles Land already asked us how they could do to get an integration like Safe Apps. Even they can copy&paste the code, it's very frequent that we add changes in the SDK that require updates in the web interface. If we can extract this to a separated package, it would be easy for other projects that reuse the Safe Apps logic to update versions.

  // If we <achieve/enable X>, then <user behavior Y changes in this way> leading to positive metrics Z. Include guesses for size of the win on specific metrics, using past launches as a baseline. 

## Links:
  
  // Include useful research, such as user tests, competitive analysis, metrics, or surveys.
  
  

# Part 2: Shaping the problem 
  
## Problem Owner

Mihail, Daniel Sa

## Non Goal(s)
  
- Adding new methods to the communicator logic
 
### Solution 1
  
**Overview**

We will extract the communicator logic to a separate package so it's easier to implement Safe Apps integration in other interfaces, so its code can be reused. Once we extract it to a separate library we will publish as a package that the web interface will use.

**Rough Scoping & Timeline**

For tackling this issue, we will need to first solve #103 instead of doing everything at once. During #103 we will already move all the communicator pieces to a limited number of files, planning to extract everything to a separate project. In this case it's easier doing it like this, because we will avoid having to work with 2 repositories which is a bigger overhead, while it's still possible to plan the separation when working in the new web codebase. Once that the web interface is ready it should just be possible to unplug the files very easily and start using the library approach. 

Once we have this package ready, we should sync with Circles, as they were planning to reuse our Safe Apps implementation to integrate with other projects.

We will also include this new library in docs to explain this new possibility

**Risk(s), Key Trade Offs & Decisions**

We won't be able to publish the library until we are almost done with the web refactor. Although from product perspective can be seen as non optimal, in terms of development is much less effort this way.

**Comment by rmeissner** [comment]
Would this be a dependency for https://github.com/safe-global/safe-pm/issues/103?

**Comment by dasanra** [comment]
> Would this be a dependency for #103?

Definitely, this has a dependency with #103 . It will be considered when working in #103 . Won't be solved in a one step, but #103 is planned so will prepare the path to tackle #104
